Configuration documentation reworked (#3583)
WeblateOrg/weblate: docs/admin/config.rst

.. versionchanged:: 3.11

   The original on/off option was changed to differentiate which strings are accepted.

Updates all repositories on a daily basis.

.. hint::

> Useful if you are not using :ref:`hooks` to update Weblate repositories automatically.

Reviewer: ```suggestion
    Useful if you are not using :ref:`hooks` to update Weblate repositories automatically.
```

.. versionadded:: 2.14

How long the authentication token and temporary password from password reset e-mails is valid for.

An amount of seconds defaulting to 172800 (2 days).

AUTH_PASSWORD_DAYS
------------------

.. versionadded:: 2.15

How many days using the same password should be allowed.

.. note::

    Password changes made prior to Weblate 2.15 will not be accounted for in this policy.

Defaults to 180 days.

.. setting:: AUTOFIX_LIST

AUTOFIX_LIST
------------

List of automatic fixes to apply when saving a string.

.. note::

Provide a fully-qualified path to the Python class that implementing the

> autofixer interface.

Reviewer: ```suggestion
    Provide a fully-qualified path to the Python class that implementing the
    autofixer interface.
```

CHECK_LIST
----------

List of quality checks to perform on a translation.

.. note::

Provide a fully-qualified path to the Python class implementing the check

> interface.

Reviewer: ```suggestion
    Provide a fully-qualified path to the Python class implementing the check
    interface.
```

:file:`ssh`
    SSH keys and configuration.
:file:`static`
    Default location for static Django files, specified by ``STATIC_ROOT``.
:file:`media`
    Default location for Django media files, specified by ``MEDIA_ROOT``.
:file:`memory`
    Translation memory data using the Whoosh engine (see :ref:`translation-memory`).
:file:`vcs`
    Version control repositories.
:file:`whoosh`
    Fulltext search index using the Whoosh engine.
:file:`backups`
    Daily backup data, see :ref:`backup-dumps`.

.. note::

    This directory has to be writable by Weblate. If if running it as uWSGI means
    the ``www-data`` user should have write access to it.

>     The easiest way to achieve this is to make the user own the directory:

Author: ```suggestion
    The easiest way to achieve this is to make the user the owner of the directory:
```

Whether to show links to share translation progress on social networks.

.. setting:: GITLAB_USERNAME

GITLAB_USERNAME
---------------

GitLab username used to send merge requests for translation updates.

.. seealso::

   :ref:`gitlab-push`,
   :ref:`lab-setup`

.. setting:: GITHUB_USERNAME

GITHUB_USERNAME
---------------

GitHub username used to send pull requests for translation updates.

.. seealso::

   :ref:`github-push`,
   :ref:`hub-setup`

.. setting:: GOOGLE_ANALYTICS_ID

GOOGLE_ANALYTICS_ID
-------------------

Google Analytics ID to turn on monitoring of Weblate using Google Analytics.

.. setting:: HIDE_REPO_CREDENTIALS

HIDE_REPO_CREDENTIALS
---------------------

Hide repository credentials from appearing in the web interface.
In case you have repository URL with user and password, Weblate will hide it
when related info is shown to users.

For example instead of ``https://user:password@git.example.com/repo.git`` it
will show just ``https://git.example.com/repo.git``. It tries to clean up VCS
error messages too in a similar manner.

.. note::

> This is turned on by default.

Reviewer: ```suggestion
    This is turned on by default.
```

> Whether to show links to share translation progress on social networks.

Author: ???

Reviewer: Whether "Share" menu is available.

Indicates whether Weblate is running behind a reverse proxy.

If set to "True", Weblate gets IP address from a header defined by :setting:`IP_BEHIND_REVERSE_PROXY`.
Ensure you are actually using a reverse proxy and that it sets this header,
otherwise users will be able to fake the IP address.

.. note::

> This is not on by default.

Reviewer: ```suggestion
    This is not on by default.
```

.. versionadded:: 3.5

URL where your Weblate instance shows its legal documents.

.. hint::

Useful if you host your legal documents outside Weblate for embedding them inside Weblate,

> please see :ref:`legal`.

Reviewer: ```suggestion
    Useful if you host your legal documents outside Weblate for embedding them inside Weblate,
    please see :ref:`legal`.
```

LICENSE_EXTRA
-------------

Additional licenses to include in the license choices.

.. note::

> Each license definition should be tuple of its short name, a long name and an URL.

Reviewer: ```suggestion
    Each license definition should be tuple of its short name, a long name and an URL.
```

LICENSE_FILTER
--------------

Optional addition of licenses to show.

.. note::

> This filter uses the short license names.

Reviewer: ```suggestion
    This filter uses the short license names.
```

LICENSE_REQUIRED
----------------

Defines whether the license attribute in :ref:`component` is required.

.. note::

> This is off by default.

Reviewer: ```suggestion
    This is off by default.
```

.. setting:: LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH

LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH
-----------------------------------------

Whether the length of a given translation should be limited.
The restriction is the length of the source string * 10 characters.

.. hint::

> Set this to ``False`` to allow longer translations (up to 10.000 characters) irrespective of source string length.

Reviewer: ```suggestion
    Set this to ``False`` to allow longer translations (up to 10.000 characters) irrespective of source string length.
```

.. note::

> Defaults to ``True``.

Reviewer: ```suggestion
    Defaults to ``True``.
```

LICENSE_REQUIRED
----------------

Defines whether the license attribute in :ref:`component` is required.

.. note::

    This is off by default.

.. setting:: LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH

LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH
-----------------------------------------

Whether the length of a given translation should be limited.
The restriction is the length of the source string * 10 characters.

.. hint::

    Set this to ``False`` to allow longer translations (up to 10.000 characters) irrespective of source string length.

.. note::

    Defaults to ``True``.

.. setting:: LOGIN_REQUIRED_URLS

LOGIN_REQUIRED_URLS
-------------------

A list of URLs you want to require logging into. (Besides the standard rules built into Weblate).

.. hint::

This allows you to password protect a whole installation using:

.. code-block:: python

> 

Reviewer: ```suggestion
    This allows you to password protect a whole installation using:

    .. code-block:: python

```

LICENSE_REQUIRED
----------------

Defines whether the license attribute in :ref:`component` is required.

.. note::

    This is off by default.

.. setting:: LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH

LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH
-----------------------------------------

Whether the length of a given translation should be limited.
The restriction is the length of the source string * 10 characters.

.. hint::

    Set this to ``False`` to allow longer translations (up to 10.000 characters) irrespective of source string length.

.. note::

    Defaults to ``True``.

.. setting:: LOGIN_REQUIRED_URLS

LOGIN_REQUIRED_URLS
-------------------

A list of URLs you want to require logging into. (Besides the standard rules built into Weblate).

.. hint::

    This allows you to password protect a whole installation using:

    .. code-block:: python

    LOGIN_REQUIRED_URLS = (

>         r'/(.*)$',

Reviewer: ```suggestion
        LOGIN_REQUIRED_URLS = (
            r'/(.*)$',
```

LICENSE_REQUIRED
----------------

Defines whether the license attribute in :ref:`component` is required.

.. note::

    This is off by default.

.. setting:: LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH

LIMIT_TRANSLATION_LENGTH_BY_SOURCE_LENGTH
-----------------------------------------

Whether the length of a given translation should be limited.
The restriction is the length of the source string * 10 characters.

.. hint::

    Set this to ``False`` to allow longer translations (up to 10.000 characters) irrespective of source string length.

.. note::

    Defaults to ``True``.

.. setting:: LOGIN_REQUIRED_URLS

LOGIN_REQUIRED_URLS
-------------------

A list of URLs you want to require logging into. (Besides the standard rules built into Weblate).

.. hint::

    This allows you to password protect a whole installation using:

    .. code-block:: python

        LOGIN_REQUIRED_URLS = (
            r'/(.*)$',
    )
    REST_FRAMEWORK["DEFAULT_PERMISSION_CLASSES"] = [
        "rest_framework.permissions.IsAuthenticated"

>     ]

Reviewer: ```suggestion
        )
        REST_FRAMEWORK["DEFAULT_PERMISSION_CLASSES"] = [
            "rest_framework.permissions.IsAuthenticated"
        ]
```

MATOMO_SITE_ID
--------------

ID of a site in Matomo (formerly Piwik) you want to track.

.. note::

> This integration does not support the Matomo Tag Manager.

Reviewer: ```suggestion
    This integration does not support the Matomo Tag Manager.
```

MATOMO_URL
----------

Full URL (including trailing slash) of a Matomo (formerly Piwik) installation you want
to use to track Weblate use. For more info about Matomo, see
<https://matomo.org/>.

.. hint::

> This integration does not support the Matomo Tag Manager.

Reviewer: ```suggestion
    This integration does not support the Matomo Tag Manager.
```

.. note::

    Many of the services need additional configuration like API keys, please check

>     their documentation for more details.

Author: There should be a link to said documentation here.

Reviewer: The docs for individual services is in https://docs.weblate.org/en/latest/admin/machine.html

>     their documentation for more details.

Author: ```suggestion
    their documentation ref: `machine` for more details.
```

------------------

Use simple language codes for default language/country combinations. For
example ``fr_FR`` translation will use the ``fr`` language code. This is usually
the desired behavior as it simplifies listing languages for these default
combinations.

Turn this off if you want to different translations for each variant.

.. setting:: SITE_TITLE

SITE_TITLE
----------

> Site title to be used for the website and e-mails.

Author: ```suggestion
Site title to be used for the website and sent e-mails.
```

.. setting:: SPECIAL_CHARS

SPECIAL_CHARS
-------------

Additional characters to include in the visual keyboard, see :ref:`visual-keyboard`.

> The default value is:

Author: This seems out of place…

STATUS_URL
----------

> The URL where your Weblate instance reports it's status.

Author: ```suggestion
The URL where your Weblate instance reports it's status.
```
Where they can be found, or where they are sent?

Reviewer: Where they can be found. On Hosted Weblate it points to https://status.weblate.org/ which is operated by third party.

.. versionadded:: 3.2.1

Automatically deletes suggestions after given number of days.
Defaults to ``None``, meaning no deletions.

.. setting:: URL_PREFIX

URL_PREFIX
----------

This settings allows you to run Weblate under some path (otherwise it relies on
being run from the webserver root).

.. note::

To use this setting, you also need to configure your server to strip this prefix.

> For example with WSGI, this can be achieved by setting ``WSGIScriptAlias``.

Reviewer: ```suggestion
    To use this setting, you also need to configure your server to strip this prefix.
    For example with WSGI, this can be achieved by setting ``WSGIScriptAlias``.
```

.. hint::

> The prefix should start with a ``/``.

Reviewer: ```suggestion
    The prefix should start with a ``/``.
```

.. note::

    This setting does not work with Django's built-in server, you would have to
    adjust :file:`urls.py` to contain this prefix.

.. setting:: VCS_BACKENDS

VCS_BACKENDS
------------

Configuration of available VCS backends.

.. note::

> Weblate tries to use all supported backends you have the tools for.

Reviewer: ```suggestion
    Weblate tries to use all supported backends you have the tools for.
```

.. hint::

> You can limit choices or add custom VCS back-ends by using this.

Reviewer: ```suggestion
    You can limit choices or add custom VCS back-ends by using this.
```

.. versionadded:: 3.10.2

Configures how deep cloning of repositories Weblate should do.

.. note::

Currently this is only supported in :ref:`vcs-git`. By default Weblate does shallow clones of the
repositories to make cloning faster and save disk space. Depending on your usage
(for example when using custom :ref:`addons`), you might want to increase

> the depth or turn off shallow clones completely by setting this to 0.

Reviewer: ```suggestion
    Currently this is only supported in :ref:`vcs-git`. By default Weblate does shallow clones of the
    repositories to make cloning faster and save disk space. Depending on your usage
    (for example when using custom :ref:`addons`), you might want to increase
    the depth or turn off shallow clones completely by setting this to 0.
```

.. hint::

> In case you get ``fatal: protocol error: expected old/new/ref, got 'shallow

Reviewer: ```suggestion
    In case you get ``fatal: protocol error: expected old/new/ref, got 'shallow
```

> <commit hash>'`` error when pushing from Weblate, turn off shallow clones completely by setting:

Reviewer: ```suggestion
    <commit hash>'`` error when pushing from Weblate, turn off shallow clones completely by setting:
```

.. versionadded:: 3.0

List of file formats available for use.

.. note::

> The default list already has the common formats.

Reviewer: ```suggestion
    The default list already has the common formats.
```
